Add `getNode` to `ModelExperimental`

j9liu opened the pull request:
This PR does the following:
- renames the node and primitive classes to `ModelExperimentalRuntimeNode` and `ModelExperimentalRuntimePrimitive`
- creates `ModelExperimentalNode` as the ME counterpart to `ModelNode`
- re-organizes `ModelExperimentalSceneGraph` to account for `node.show`
- re-enables `ModelExperimental` on [CZML Model - Node Transformations](http://localhost:8080/Apps/Sandcastle/index.html?src=CZML%20Model%20-%20Node%20Transformations.html) and [Time Dynamic Wheels](http://localhost:8080/Apps/Sandcastle/index.html?src=Time%20Dynamic%20Wheels.html)
- re-enables `ModelExperimental` in `ModelVisualizerSpec`

![node](https://user-images.githubusercontent.com/32226860/178578158-ab8a480c-4b31-4773-8836-92c237db998d.gif)

Note: `ModelNode` has no way to let the user re-enable the glTF animation if they want to, and also no method for handling `node.matrix = undefined`. In `ModelExperimentalNode` I made this a method of "resetting" the node such that it can be animated by the glTF again. Here's a [Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=1VZtT+s2FP4rVr+sTMVJS4HSC2hcLrBNcIdotUm7uUImOW28OnZlOwU28d93bCdpCmWbdD+tQiG2z9vznBcnVdJYsuLwCJqcEAmP5BwMLwv6q9/rJp3Ur8+VtIxL0EmnR/5KJCEmV6XIziQvmIUxsbqEXiJfdj4kMpHBIk2FShc0LbUGaae8APRR2f+5FJzJT6hLZ1oVPxk1Ooj7XWc56QziQX+339+Nj6aDeDw4HO8d0rh/EB8Mh4f7e8PRwWh4tHf4e9JJZPAXReRGZSBMIlMPqfArdPfFWfQBE2LhyY7RfBV0RqYaY5gLQFBBAnVBQIpSs1KmlitJuju1OmL2h95Tt95z8UaTJaQmQjQsCnFEV8LOrhXLQEe1u9pbNBfTyze7dI4aTSDuF7feHb31ykF2/1/8eXi+QjhZcCkR4HnONEutT9s3IaQ0wr8JK5YCPMxAcBSyecNk9XaPr4jk4V0gMybMPyD56tMZcpgvX9Xkj8AyLue33Kb5nRIhwurshtmcWnWHEkyabn8U73ijwXccCiXYnfEnyC41K2CqUXamdLGuy2bLUCxeJoKcumx0rgC7gFmlq1qVStu8gpt0HsHYdVkKqCoRVw3dbYpLLXokBz7PbY9ItH6nlP2MR1VKQsBLZbhXbaI8Z9riG5N7vns+wVwDmCpju/3BHo0Ph8OD/lHF/HBI4/147zA+qDaCT/ceIiXVEKAmRXx0qXmBLldgqIZCreAM2Q7Z8ngwkvfkWZZVcdSZcQoXT0tAGRwDTPiQXX90m7JDHsbu0RSK94I51fxpvCU1+atKaCeoVbc1ca36w6JqrSrLF0LwpVE8o79dTUbDdrm+rZWmeNfBcuzlsrhFWTHhf+I07A9G61P25E4nWEx4MojxV5297LQT4CEj3Sx7vkWCuAFqc5DddZ96iaZZ6xmLsWmGxaoWZ7bB/gb5e22Es6fN2PZuGu7v9P5NZre/IRTQPKhSOl+TZQ4aqEbZ0pDvyX5DYnipOaj1cGZhMYURiSh8VbkSbAoG4S7HGxXWCF/jEb27uL04mzY8V3MmPPmsu6XV3A+vkWnODcE/RnKWLsgDpKw0QDATOIPDpCYStUimwMjvcEqxFaC0s9gyw6VXcZOe1tuhmb3ySYXz3q3Mx+fPqP0FqYIZ3rHZ1w8bbbAhRV3YbiziVj09/fB0GF8cixMms5QZi1cKsjZVSjwwfQOyDPVjAtdv5eYI7WNprZJ4609wctU3E3EsuXvfz+5ec3F0V0yUm5NqA9wcPL9ozcWcdEIWkP22HvE6tPCtvh5wofWHflT4zhM+t90t5byehd2Yxj1SPfZbpYX0EEQO2102vAfJ5iPmP1DUXH/fztFkgfcCWr6XkC7u/1Bc2vv+/4+0Tq9zbOyzgNPa6w+8WOId6cZ7F78hLOA3BH74mOihTBdgaWpM3Z/HUVv1OOMrwrOTLd+gJBXMGDyZlcKP3KRzehyh/BtVofyw+2UFWrBnJ5b3T6/DJqX0OMLldk0bOueV5b8B) that shows how that behavior will work.

Additional [ModelExperimental Node Explorer Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=xVn7c9s2Ev5XUN8PphqZkh9pe4rju4uTtJ25NLnYyVSKMh2IhCTUJMABQNm6jv73W7wo8CFnnIvtTDIhHovd/fbDYgENBkgtqUQJZmhGULLEbEFSpDjCbI0W2eVrlPOUZFOWcCaVbXwQGXqOpntxPIC/FzgvMvISKzwwo3JwTiQt8zeYua8/4DNeZLPp3rMpm7LBAH2QxOo1ElpdzgVBZD4niaIrkq2RIqCuIOKAwRQkl/xaS97Bip8FL1n6kSxpkpF6KzDGLrii5PqNMeU5+mvKENJKf8M5GSGQInPKSNrX/dqOS4GZzLCinI3QHGeSVEPvuersv0hwRsJOpdeYc5Gb6XKE/tpA/8aYZEGLrxhPrnipYpibXEWViT0PYn0NBGhilPC8KBVEsBAcwFNrJIgqBaNsAYATtMJZSSSSigu8IAikTXdSCkGYMm4jBn63rbAovHPLRtqJyiTj0nSvbtB0z3TPS5YY+6KehRYhi3kDAkC+Wi9ujD0L5XxoagK+082kcxR95zxw4Ysaa37yIp97lWGtuGwneWIE0ywFfh+hYTzsd46NbxmbNMeE484vBKcQrl3D76hKlrsG3/Msa45JzT2w8rDdO+7sndR7N8/8Vzcxd+PqBDf2P0vE3Qib2TB3y+8lgTRxLajCs4y0qE2Bx/CZlglBlKV0RdMSZ57hmtml1COG4B9+nbJPAUtd7BxHa53jrs5J1dkIU6vfxKfVqwNTddqI1JvjetPq+xyDG69wsoy2u6hwbP3C/tzuTVT0PXUXRI26NmQVnF078FPx2QfT8ULWlzKoB+tB+BJORFJlHEjxrMxnRPgZt6iCvfbEiNV0bjQxNg12dJHC+UIAN0QykuvExueGBpWqL+a3AL8aESAubQSdRkaufVjOsYADjGJ2HFkfGu46/vVvGRzfNjjRYxUgd/HFM7LTEZthl4WAEATeOK4bamsmt3xqbIr+rnGzws5Rs0mcXxWozoT/lFgROMc4i+eC5y2LwOavgsPstv8/qHZL9zu7x93d9RAOfCHWOsChZwbmS0NgPOMrEpKdMl2rIQlgQI6ELSTojd5sUMzhosioLea0qE60d8PGrnYbOG61N2biiYlMUB1tY+brIlMH3bYjdnKjG8MahL6YU5xnM6wpnPKk1Ns/hsT3ymaCF+tf0wh2tJ0z3eviiwZu/UKfKWwhQ0icVK9ROZLGdvlo+kBLYtrnnCkMIDttVk4mhPkShojYNINll4QulgrGj54O4U88rAS5oAs41J63SWngf0kWghBpMD44PDqOhz+enPxw+HeD38lJPHw6PP5x+INpWiX2xA0KaxvNrYZLnzRlvGxsu0v+mt6Q9LWAA9yotNaZ1W/LHz1/zAdqQaFBIS4Ezam+BcgYp2kUHHYmDK9ugKpUxxJnxuWfMzWP3MlTCqh//PXAcSZwahQ2/DBlNC/zd+BIdkH/C2X64dFP7sTxZuaWi+DGGrZKTqVhXgzbigVHs5nVqHITQEbgbaBt2yzqzskJ57m9AZl7ViAKvBGAlqGXhUYmEFx2UeCEnJuFzqs5tRoZJPw5exQP0fcIPF7GOb6xNsYzfRmCiFwUSyJILCA8pew7YwHUUiqAmxHLdbeuUxQ7wLTdL6lUmCWayAK0DOOnlWfOBwiTsb9Trx2tWe4YtmWfsVzx99ANNIyOjmEz9GoihabVToGDo1DAOphxfvUvFXmIrBlVwbuLuPpqHDn7+lZr37gNEPescC8MLIGCB7Qpm3rhdsZSBIRGvDCVb+iC6wIn3s7+hCtwfEXW0sXqDyP9Yq2L5B7EsKjs3jLPF9G98JLicnTQAxcQcgN1m5/eD4dAP1RLya4KsVm1BdewxuXLXRuCxbdXiM0WJ6PT+v1p+Dn26iM/dgF4JVgquK4Dapc29b4BUCMnVQP7Uh+enGVrhM0jgnSgqyVWKBX4GgVpBV0QpeyVGKRMqVktpC8OGP3Jqb8OX9MsQ0u80mF0LxRxt4ULOIJflEpxBrn/Aq796Dd95sL5qURJtqeoq5YraLf32mqjwIGlZaM23r0KS90T69cFnV3CcrkGjCCy4MxUAXoxm2rtO49Es7WpEtb+eaCqPtzRzzqufV69tvHtTBKx0tez7sqhV02W5QySF52RIF/CRvvYuDd8NRjdd34DkT2TsMv6vQanO6Y4SsfWizjJOCNRqGqz/Qwm1lKQqYfyMlMU4I2+oLC2Dz0qzc7G4lFvO761bWM/N2YQDhtVuzsSIbio/L+GCodfx1BGCeWGnlUXrSnb6++dSrXOyJlf/J80L7hQ+pCN4nigSF5A1UbkYFYmV0TFiZRbS/7my7AA7RlOrhbmIW6ExGKGo5OjPvL/hvFPIcYFbCnzDHJS3ATdMy5SIg7siVUf3LRUUwZsDg1Y6aoZCscDnNEFg2qApmlG2loPFC9G6Kim2Q/NOOzwvD7qVJ8OQsBOU7pCNH3eUQeiJMNSwsi8zEzRMd07Ox3A/JZoxs158xYsz/BaT1senv3bdsZxfDqAZrdkVeJW8Ts1DylnW5dO1Yyn67OQaKdKnNVT/qlKYVNmssAMVj0C41Os8AHcSYxrGU2utkeGPjGaj6SwL75rdm0CsypFWkVtcXtaNWX/gfYP9tEI7T/Zh1WeAOgg11otkGj4M1Bp3eVBw2fAoGbGiko60++3DUtaPgBUZ7+31rf9LW8tOdW6IKBC6HwM0EJtBa2DY/2Jb+BTf0lFCviEOuawgb5J+6Pac2TfngUfCpgHQ/tGzX4X3DUDNNRaFXCxI8odetrO3y+u48fFdfxAuI4fGtfJ4+I6eSBcJ1+H6zfJhf71wyfCqn3XLFgJdqdA5Me/KYGqRbvY88sjsKf52HiPBGo99p89HLTvHhFa+077AMC6X0seENb3jwirfvt6CFTtr02Plu/MK69PdrZx10xnpXakuQv/9PvNGGNXvK9yzhHjDoRxPyfcI1X8b5BnD4Tj+PFwHN87juOHw3HyeDhO7h3Huxdp0KxdYaEd3HHDi/H/AA) for testing.

ptrgags commented on Source/Scene/ModelExperimental/ModelExperimentalSceneGraph.js:
```diff
@@ -424,6 +425,12 @@ function traverseSceneGraph(sceneGraph, node, transformToRoot) {
+  // Create and store the public version of the runtime node.
+  const model = sceneGraph._model;
+  const publicNode = new ModelExperimentalNode(model, runtimeNode);
+  const name = publicNode.name;
+  model._nodesByName[name] = publicNode;
```
what should be the behavior if `name` is undefined? (name is not a required field). Do we skip creating a public node since there's no way to reference it?

j9liu replied:
`Model` didn't account for this, but it makes sense to ignore the node if we can't reference it by name.

ptrgags commented on Source/Scene/ModelExperimental/ModelExperimentalSceneGraph.js:
```diff
@@ -672,13 +679,54 @@ ModelExperimentalSceneGraph.prototype.updateJointMatrices = function () {
-    }
+/**
+ * Recursively traverse through the runtime nodes in the scene graph
+ * using depth-first post-order traversal to perform a callback on
```
Just noticed this, I think this wording is clearer since "post-order" is modifying "depth-first traversal"

```suggestion
 * using a post-order depth-first traversal to perform a callback on
```

ptrgags commented on Specs/Scene/ModelExperimental/ModelExperimentalRuntimeNodeSpec.js:
```diff
@@ -0,0 +1,502 @@
+  ) {
+    originalTransform = defaultValue(originalTransform, transform);
+
+    expect(Matrix4.equals(runtimeNode.transform, transform)).toBe(true);
```
Can these be written like this?

```
expect(runtimeNode.transform).toEqual(transform);
```

ptrgags commented on Source/Scene/ModelExperimental/ModelExperimentalAnimationChannel.js:
```diff
@@ -275,17 +276,17 @@ ModelExperimentalAnimationChannel.prototype.animate = function (time) {
         : spline.wrapTime(time);
       morphWeights[i] = spline.evaluate(localAnimationTime);
     }
+  } else if (runtimeNode.userAnimated) {
+    // If the node is being animated externally, ignore the glTF animation.
+    return;
```
Just to clarify, the weights still need to be updated even if the user modified the nodes?

j9liu replied:
Yes, I believe the animation of morph weights in `Model` happens whether or not the node's transform has been modified by the user. The `translation`, `rotation`, and `scale` properties are ignored, but the weights are unrelated.

ptrgags reviewed (commented):
@j9liu I had a few comments on the code. I still need to try running this locally.

ptrgags commented:
When running [the ME node explorer sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=xVn7c9s2Ev5XUN8PphqZkh9pe4rju4uTtJ25NLnYyVSKMh2IhCTUJMABQNm6jv73W7wo8CFnnIvtTDIhHovd/fbDYgENBkgtqUQJZmhGULLEbEFSpDjCbI0W2eVrlPOUZFOWcCaVbXwQGXqOpntxPIC/FzgvMvISKzwwo3JwTiQt8zeYua8/4DNeZLPp3rMpm7LBAH2QxOo1ElpdzgVBZD4niaIrkq2RIqCuIOKAwRQkl/xaS97Bip8FL1n6kSxpkpF6KzDGLrii5PqNMeU5+mvKENJKf8M5GSGQInPKSNrX/dqOS4GZzLCinI3QHGeSVEPvuersv0hwRsJOpdeYc5Gb6XKE/tpA/8aYZEGLrxhPrnipYpibXEWViT0PYn0NBGhilPC8KBVEsBAcwFNrJIgqBaNsAYATtMJZSSSSigu8IAikTXdSCkGYMm4jBn63rbAovHPLRtqJyiTj0nSvbtB0z3TPS5YY+6KehRYhi3kDAkC+Wi9ujD0L5XxoagK+082kcxR95zxw4Ysaa37yIp97lWGtuGwneWIE0ywFfh+hYTzsd46NbxmbNMeE484vBKcQrl3D76hKlrsG3/Msa45JzT2w8rDdO+7sndR7N8/8Vzcxd+PqBDf2P0vE3Qib2TB3y+8lgTRxLajCs4y0qE2Bx/CZlglBlKV0RdMSZ57hmtml1COG4B9+nbJPAUtd7BxHa53jrs5J1dkIU6vfxKfVqwNTddqI1JvjetPq+xyDG69wsoy2u6hwbP3C/tzuTVT0PXUXRI26NmQVnF078FPx2QfT8ULWlzKoB+tB+BJORFJlHEjxrMxnRPgZt6iCvfbEiNV0bjQxNg12dJHC+UIAN0QykuvExueGBpWqL+a3AL8aESAubQSdRkaufVjOsYADjGJ2HFkfGu46/vVvGRzfNjjRYxUgd/HFM7LTEZthl4WAEATeOK4bamsmt3xqbIr+rnGzws5Rs0mcXxWozoT/lFgROMc4i+eC5y2LwOavgsPstv8/qHZL9zu7x93d9RAOfCHWOsChZwbmS0NgPOMrEpKdMl2rIQlgQI6ELSTojd5sUMzhosioLea0qE60d8PGrnYbOG61N2biiYlMUB1tY+brIlMH3bYjdnKjG8MahL6YU5xnM6wpnPKk1Ns/hsT3ymaCF+tf0wh2tJ0z3eviiwZu/UKfKWwhQ0icVK9ROZLGdvlo+kBLYtrnnCkMIDttVk4mhPkShojYNINll4QulgrGj54O4U88rAS5oAs41J63SWngf0kWghBpMD44PDqOhz+enPxw+HeD38lJPHw6PP5x+INpWiX2xA0KaxvNrYZLnzRlvGxsu0v+mt6Q9LWAA9yotNaZ1W/LHz1/zAdqQaFBIS4Ezam+BcgYp2kUHHYmDK9ugKpUxxJnxuWfMzWP3MlTCqh//PXAcSZwahQ2/DBlNC/zd+BIdkH/C2X64dFP7sTxZuaWi+DGGrZKTqVhXgzbigVHs5nVqHITQEbgbaBt2yzqzskJ57m9AZl7ViAKvBGAlqGXhUYmEFx2UeCEnJuFzqs5tRoZJPw5exQP0fcIPF7GOb6xNsYzfRmCiFwUSyJILCA8pew7YwHUUiqAmxHLdbeuUxQ7wLTdL6lUmCWayAK0DOOnlWfOBwiTsb9Trx2tWe4YtmWfsVzx99ANNIyOjmEz9GoihabVToGDo1DAOphxfvUvFXmIrBlVwbuLuPpqHDn7+lZr37gNEPescC8MLIGCB7Qpm3rhdsZSBIRGvDCVb+iC6wIn3s7+hCtwfEXW0sXqDyP9Yq2L5B7EsKjs3jLPF9G98JLicnTQAxcQcgN1m5/eD4dAP1RLya4KsVm1BdewxuXLXRuCxbdXiM0WJ6PT+v1p+Dn26iM/dgF4JVgquK4Dapc29b4BUCMnVQP7Uh+enGVrhM0jgnSgqyVWKBX4GgVpBV0QpeyVGKRMqVktpC8OGP3Jqb8OX9MsQ0u80mF0LxRxt4ULOIJflEpxBrn/Aq796Dd95sL5qURJtqeoq5YraLf32mqjwIGlZaM23r0KS90T69cFnV3CcrkGjCCy4MxUAXoxm2rtO49Es7WpEtb+eaCqPtzRzzqufV69tvHtTBKx0tez7sqhV02W5QySF52RIF/CRvvYuDd8NRjdd34DkT2TsMv6vQanO6Y4SsfWizjJOCNRqGqz/Qwm1lKQqYfyMlMU4I2+oLC2Dz0qzc7G4lFvO761bWM/N2YQDhtVuzsSIbio/L+GCodfx1BGCeWGnlUXrSnb6++dSrXOyJlf/J80L7hQ+pCN4nigSF5A1UbkYFYmV0TFiZRbS/7my7AA7RlOrhbmIW6ExGKGo5OjPvL/hvFPIcYFbCnzDHJS3ATdMy5SIg7siVUf3LRUUwZsDg1Y6aoZCscDnNEFg2qApmlG2loPFC9G6Kim2Q/NOOzwvD7qVJ8OQsBOU7pCNH3eUQeiJMNSwsi8zEzRMd07Ox3A/JZoxs158xYsz/BaT1senv3bdsZxfDqAZrdkVeJW8Ts1DylnW5dO1Yyn67OQaKdKnNVT/qlKYVNmssAMVj0C41Os8AHcSYxrGU2utkeGPjGaj6SwL75rdm0CsypFWkVtcXtaNWX/gfYP9tEI7T/Zh1WeAOgg11otkGj4M1Bp3eVBw2fAoGbGiko60++3DUtaPgBUZ7+31rf9LW8tOdW6IKBC6HwM0EJtBa2DY/2Jb+BTf0lFCviEOuawgb5J+6Pac2TfngUfCpgHQ/tGzX4X3DUDNNRaFXCxI8odetrO3y+u48fFdfxAuI4fGtfJ4+I6eSBcJ1+H6zfJhf71wyfCqn3XLFgJdqdA5Me/KYGqRbvY88sjsKf52HiPBGo99p89HLTvHhFa+077AMC6X0seENb3jwirfvt6CFTtr02Plu/MK69PdrZx10xnpXakuQv/9PvNGGNXvK9yzhHjDoRxPyfcI1X8b5BnD4Tj+PFwHN87juOHw3HyeDhO7h3Huxdp0KxdYaEd3HHDi/H/AA) when I first load the page and adjust the translations for the whole model, it doesn't seem to translate, though I notice some sort of clipping:

![image](https://user-images.githubusercontent.com/8422414/178817602-8d690899-6283-48b8-a1f0-6e9f004b2a95.png)

j9liu commented:
> When running [the ME node explorer sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=xVn7c9s2Ev5XUN8PphqZkh9pe4rju4uTtJ25NLnYyVSKMh2IhCTUJMABQNm6jv73W7wo8CFnnIvtTDIhHovd/fbDYgENBkgtqUQJZmhGULLEbEFSpDjCbI0W2eVrlPOUZFOWcCaVbXwQGXqOpntxPIC/FzgvMvISKzwwo3JwTiQt8zeYua8/4DNeZLPp3rMpm7LBAH2QxOo1ElpdzgVBZD4niaIrkq2RIqCuIOKAwRQkl/xaS97Bip8FL1n6kSxpkpF6KzDGLrii5PqNMeU5+mvKENJKf8M5GSGQInPKSNrX/dqOS4GZzLCinI3QHGeSVEPvuersv0hwRsJOpdeYc5Gb6XKE/tpA/8aYZEGLrxhPrnipYpibXEWViT0PYn0NBGhilPC8KBVEsBAcwFNrJIgqBaNsAYATtMJZSSSSigu8IAikTXdSCkGYMm4jBn63rbAovHPLRtqJyiTj0nSvbtB0z3TPS5YY+6KehRYhi3kDAkC+Wi9ujD0L5XxoagK+082kcxR95zxw4Ysaa37yIp97lWGtuGwneWIE0ywFfh+hYTzsd46NbxmbNMeE484vBKcQrl3D76hKlrsG3/Msa45JzT2w8rDdO+7sndR7N8/8Vzcxd+PqBDf2P0vE3Qib2TB3y+8lgTRxLajCs4y0qE2Bx/CZlglBlKV0RdMSZ57hmtml1COG4B9+nbJPAUtd7BxHa53jrs5J1dkIU6vfxKfVqwNTddqI1JvjetPq+xyDG69wsoy2u6hwbP3C/tzuTVT0PXUXRI26NmQVnF078FPx2QfT8ULWlzKoB+tB+BJORFJlHEjxrMxnRPgZt6iCvfbEiNV0bjQxNg12dJHC+UIAN0QykuvExueGBpWqL+a3AL8aESAubQSdRkaufVjOsYADjGJ2HFkfGu46/vVvGRzfNjjRYxUgd/HFM7LTEZthl4WAEATeOK4bamsmt3xqbIr+rnGzws5Rs0mcXxWozoT/lFgROMc4i+eC5y2LwOavgsPstv8/qHZL9zu7x93d9RAOfCHWOsChZwbmS0NgPOMrEpKdMl2rIQlgQI6ELSTojd5sUMzhosioLea0qE60d8PGrnYbOG61N2biiYlMUB1tY+brIlMH3bYjdnKjG8MahL6YU5xnM6wpnPKk1Ns/hsT3ymaCF+tf0wh2tJ0z3eviiwZu/UKfKWwhQ0icVK9ROZLGdvlo+kBLYtrnnCkMIDttVk4mhPkShojYNINll4QulgrGj54O4U88rAS5oAs41J63SWngf0kWghBpMD44PDqOhz+enPxw+HeD38lJPHw6PP5x+INpWiX2xA0KaxvNrYZLnzRlvGxsu0v+mt6Q9LWAA9yotNaZ1W/LHz1/zAdqQaFBIS4Ezam+BcgYp2kUHHYmDK9ugKpUxxJnxuWfMzWP3MlTCqh//PXAcSZwahQ2/DBlNC/zd+BIdkH/C2X64dFP7sTxZuaWi+DGGrZKTqVhXgzbigVHs5nVqHITQEbgbaBt2yzqzskJ57m9AZl7ViAKvBGAlqGXhUYmEFx2UeCEnJuFzqs5tRoZJPw5exQP0fcIPF7GOb6xNsYzfRmCiFwUSyJILCA8pew7YwHUUiqAmxHLdbeuUxQ7wLTdL6lUmCWayAK0DOOnlWfOBwiTsb9Trx2tWe4YtmWfsVzx99ANNIyOjmEz9GoihabVToGDo1DAOphxfvUvFXmIrBlVwbuLuPpqHDn7+lZr37gNEPescC8MLIGCB7Qpm3rhdsZSBIRGvDCVb+iC6wIn3s7+hCtwfEXW0sXqDyP9Yq2L5B7EsKjs3jLPF9G98JLicnTQAxcQcgN1m5/eD4dAP1RLya4KsVm1BdewxuXLXRuCxbdXiM0WJ6PT+v1p+Dn26iM/dgF4JVgquK4Dapc29b4BUCMnVQP7Uh+enGVrhM0jgnSgqyVWKBX4GgVpBV0QpeyVGKRMqVktpC8OGP3Jqb8OX9MsQ0u80mF0LxRxt4ULOIJflEpxBrn/Aq796Dd95sL5qURJtqeoq5YraLf32mqjwIGlZaM23r0KS90T69cFnV3CcrkGjCCy4MxUAXoxm2rtO49Es7WpEtb+eaCqPtzRzzqufV69tvHtTBKx0tez7sqhV02W5QySF52RIF/CRvvYuDd8NRjdd34DkT2TsMv6vQanO6Y4SsfWizjJOCNRqGqz/Qwm1lKQqYfyMlMU4I2+oLC2Dz0qzc7G4lFvO761bWM/N2YQDhtVuzsSIbio/L+GCodfx1BGCeWGnlUXrSnb6++dSrXOyJlf/J80L7hQ+pCN4nigSF5A1UbkYFYmV0TFiZRbS/7my7AA7RlOrhbmIW6ExGKGo5OjPvL/hvFPIcYFbCnzDHJS3ATdMy5SIg7siVUf3LRUUwZsDg1Y6aoZCscDnNEFg2qApmlG2loPFC9G6Kim2Q/NOOzwvD7qVJ8OQsBOU7pCNH3eUQeiJMNSwsi8zEzRMd07Ox3A/JZoxs158xYsz/BaT1senv3bdsZxfDqAZrdkVeJW8Ts1DylnW5dO1Yyn67OQaKdKnNVT/qlKYVNmssAMVj0C41Os8AHcSYxrGU2utkeGPjGaj6SwL75rdm0CsypFWkVtcXtaNWX/gfYP9tEI7T/Zh1WeAOgg11otkGj4M1Bp3eVBw2fAoGbGiko60++3DUtaPgBUZ7+31rf9LW8tOdW6IKBC6HwM0EJtBa2DY/2Jb+BTf0lFCviEOuawgb5J+6Pac2TfngUfCpgHQ/tGzX4X3DUDNNRaFXCxI8odetrO3y+u48fFdfxAuI4fGtfJ4+I6eSBcJ1+H6zfJhf71wyfCqn3XLFgJdqdA5Me/KYGqRbvY88sjsKf52HiPBGo99p89HLTvHhFa+077AMC6X0seENb3jwirfvt6CFTtr02Plu/MK69PdrZx10xnpXakuQv/9PvNGGNXvK9yzhHjDoRxPyfcI1X8b5BnD4Tj+PFwHN87juOHw3HyeDhO7h3Huxdp0KxdYaEd3HHDi/H/AA) when I first load the page and adjust the translations for the whole model, it doesn't seem to translate, though I notice some sort of clipping:

This _should_ be the same behavior as `Model` -- can you double check? It's `Armature` that actually moves the whole model.

ptrgags commented:
@j9liu ah, you're right, `Armature` works while `CesiumMan` reacts weirdly even in the old Model.

ptrgags commented:
the sandcastles seem to be running fine!

For the unit tests, I've seen that sporadic `afterAll` test failure when I run locally. Could you merge in `main` to get changes from #10548, that would make it easier to debug.

j9liu commented:
@ptrgags I looked at the ["Throwing Exceptions"](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/CodingGuide#throwing-exceptions) part of the documentation guide and I think that the way I implemented the error was wrong. I changed it to a `console.log` to keep it consistent with 3D Tiles -- when they don't load, they `console.log` their errors.

j9liu commented:
Still getting this error in Travis... I'm going to disable the unit tests involving this URL for the time being and see if that makes Travis pass.
![image](https://user-images.githubusercontent.com/32226860/179017526-90e9a3e8-832a-44fe-a8cb-12a8614bfe03.png)

ptrgags commented on Source/Scene/GltfLoader.js:
```diff
@@ -130,6 +130,13 @@ const GltfLoaderState = {
    */
   FAILED: 7,
+  /**
+   * If the loader gets unloaded, it switches to the destroyed state.
```
```suggestion
   * If unload() is called, the loader switches to the unloaded state.
```

ptrgags reviewed (commented):
@j9liu the changes look good, and this should also avoid the Travis failures for now.

I had one comment about documentation wording, but I'll fix that through the UI and merge.
